mongo/integration/collection_test.go

```diff
@@ -778,15 +779,140 @@ func TestCollection(t *testing.T) {
 		})
 		mt.Run("options", func(mt *mtest.T) {
 			initCollection(mt, mt.Coll)
-			opts := options.FindOne().SetComment("here's a query for ya")
-			res, err := mt.Coll.FindOne(mtest.Background, bson.D{{"x", 1}}, opts).DecodeBytes()
+
+			// Create an index for Hint, Max, and Min options
+			indexView := mt.Coll.Indexes()
+			indexName, err := indexView.CreateOne(context.Background(), mongo.IndexModel{
+				Keys: bson.D{{"x", 1}},
+			})
+			assert.Nil(mt, err, "CreateOne error: %v", err)
+
+			mt.ClearEvents()
+			expectedCollation := options.Collation{Locale: "en_US"}
+			expectedComment := "here's a query for ya"
+			expectedMax := bson.D{{"x", int32(5)}}
+			expectedMin := bson.D{{"x", int32(0)}}
+			expectedProjection := bson.D{{"x", int32(1)}}
+			expectedSort := bson.D{{"x", int32(1)}}
+			// SetCursorType is excluded because tailable cursors can't be used with limit -1,
+			// which all FindOne operations set, and the nontailable setting isn't passed to the
+			// operation layer
+			// SetMaxAwaitTime affects the cursor and not the server command, so it can't be checked
+			opts := options.FindOne().
+				SetAllowPartialResults(true).
+				SetBatchSize(2).
+				SetCollation(&expectedCollation).
+				SetComment(expectedComment).
+				SetHint(indexName).
+				SetMax(expectedMax).
+				SetMaxTime(1 * time.Second).
+				SetMin(expectedMin).
+				SetNoCursorTimeout(false).
+				SetOplogReplay(false).
+				SetProjection(expectedProjection).
+				SetReturnKey(false).
+				SetShowRecordID(false).
+				SetSkip(0).
+				SetSort(expectedSort)
+			res, err := mt.Coll.FindOne(mtest.Background, bson.D{}, opts).DecodeBytes()
 			assert.Nil(mt, err, "FindOne error: %v", err)
 
 			x, err := res.LookupErr("x")
 			assert.Nil(mt, err, "x not found in document %v", res)
 			assert.Equal(mt, bson.TypeInt32, x.Type, "expected x type %v, got %v", bson.TypeInt32, x.Type)
 			got := x.Int32()
 			assert.Equal(mt, int32(1), got, "expected x value 1, got %v", got)
+
+			started := mt.GetStartedEvent()
+			assert.NotNil(mt, started, "expected CommandStartedEvent, got nil")
+
+			aprVal, err := started.Command.LookupErr("allowPartialResults")
```

Comment:
Rather than checking every field manually, is it possible to declare a `bson.M` with the expected options and use the `compareDocs` helper?

Reply:
my issue with that is that there are more things in the command than just the expected options. I was also considering creating a struct for the expected values and comparing one that's unmarshaled from actual to an expected one, but for that solution it's hard to tell what's off if they're not equal

Reply:
We use the `compareDocs` helper in spec tests when we do command monitoring expectations, so it should already be ignoring extra fields in the `actual` parameter.